src/compiler/extensions.ts

```diff
@@ -0,0 +1,379 @@
+         * @param {string} err The error message to report
+         * @param {Node} span The node on which to position the error
+         */
+        (level: DiagnosticCategory, err: string, span: Node): void;
+        /**
+         * @param {DiagnosticCategory} level The error level to report this error as (Message, Warning, or Error)
+         * @param {string} err The error message to report
+         * @param {number} start The start position of the error span
+         * @param {number} length The length of the error span
+         */
+        (level: DiagnosticCategory, err: string, start: number, length: number): void;
+        /**
+         * @param {DiagnosticCategory} level The error level to report this error as (Message, Warning, or Error)
+         * @param {string} shortname A short code uniquely identifying the error within the lint
+         * @param {string} err The error message to report
+         */
+        (level: DiagnosticCategory, shortname: string, err: string): void;
+        /**
+         * @param {DiagnosticCategory} level The error level to report this error as (Message, Warning, or Error)
+         * @param {string} shortname A short code uniquely identifying the error within the lint
+         * @param {string} err The error message to report
+         * @param {Node} span The node on which to position the error
+         */
+        (level: DiagnosticCategory, shortname: string, err: string, span: Node): void;
+        /**
+         * @param {DiagnosticCategory} level The error level to report this error as (Message, Warning, or Error)
+         * @param {string} shortname A short code uniquely identifying the error within the lint
+         * @param {string} err The error message to report
+         * @param {number} start The start position of the error span
+         * @param {number} length The length of the error span
+         */
+        (level: DiagnosticCategory, shortname: string, err: string, start: number, length: number): void;
+    };
+
+    /*
+    * Walkers call stop to halt recursion into the node's children
+    * Walkers call error to add errors to the output.
+    */
+    export interface LintWalker {
+        /**
+         * @param {Node} node The current node being visited (starts at every SourceFile and recurs into their children)
+         * @param {LintErrorMethod} error A callback to add errors to the output
+         * @returns boolean true if this lint no longer needs to recur into the active node
+         */
+        visit(node: Node, error: LintErrorMethod): boolean | void;
+        /**
+         * Yar
+         * @param {Node} node The current node which has just finished being visited
+         * @param {LintErrorMethod} error A callback to add errors to the output
+         */
+        afterVisit?(node: Node, error: LintErrorMethod): void;
+    }
+
+    export interface BaseProviderStatic {
+        readonly ["extension-kind"]: ExtensionKind;

```

Comment:
Is it enough to say that it mimics the style of the key? :3
IDK, whenever I've seen stringy enumerations in JS they've either been `ALL_CAPS` or `"less-mad"`. I prefer the latter.